server/server/data_loader/arangoload.py

```diff
@@ -83,14 +83,28 @@ def collect_data(repo_dir: Path, repo_addr: str):
             update_data(repo, repo_addr)
 
 
-def process_root_languages(structure_dir):
-    language_file = structure_dir / 'language.json'
-    languages = json_load(language_file)
+def process_root_languages(language_file: Path) -> Dict[str, str]:
+    languages: List[dict] = json_load(language_file)
     data = {}
-    for lang in languages:
-        uid = lang['uid']
-        if 'contains' in lang:
-            data.update({sutta_id: uid for sutta_id in lang['contains']})
+    for language in languages:
+        lang_iso = language['root_lang_iso']
+        if 'contains' in language:
+            data.update({uid: lang_iso for uid in language['contains']})
+    return data
+
+
+def process_extra_info_file(extra_info_file: Path) -> Dict[str, Dict[str, str]]:
+    """
+    Method to process super_extra_info.json and text_extra_info.json files
+
+    Args:
+        extra_info_file - path to the file
+    """
+    info = json_load(extra_info_file)
+    data = {}
+    for item in info:
+        uid = item.get('uid')
+        data.update({uid: item})
```

Comment:
I am unsure why you are using `data.update({uid: item})` instead of simple assignment `data[uid] = item` or a dict comprehension.

Reply:
Indeed, thanks!
I missed that.
Fixed in the next commit!